src/v/cluster/cluster_discovery.cc

```diff
@@ -185,4 +366,37 @@ cluster_discovery::get_node_index_in_seed_servers() {
       boost::numeric_cast<int32_t>(std::distance(seed_servers.cbegin(), it))};
 }
 
+bool cluster_discovery::is_amnesiac_seed_server() const {
+    vassert(
+      _cluster_bootstrap_info_fetched,
+      "Cluster bootstrap info not fetched yet");
+
+    bool failed = false;
+    std::optional<model::cluster_uuid> peer_cluster_uuid;
+    for (auto& peer : _cluster_bootstrap_info) {
+        if (peer.second.cluster_uuid) {
+            if (peer_cluster_uuid) {
```

Comment:
nit: IMO easier to follow and reduces scoping written as:

```
if (peer_cluster_uuid == std::nullopt) {
    peer_cluster_uuid = peer.second.cluster_uuid;
    continue;
}
if (*peer_cluster_uuid != *peer.second.cluster_uuid) {
    ...
}

```

Reply:
```
if (peer_cluster_uuid == std::nullopt) {
    ...
    continue;
}
```
won't work, there's more in the loop body

Reply:
Or how about something like:

```
for (const auto& [addr, reply] : _cluster_bootstrap_info) {
    check_configuration_match(failed, addr, reply);
    if (!reply.cluster_uuid) {
        continue;
    }
    if (peer_cluster_uuid == std::nullopt) {
        *peer_cluster_uuid = reply.cluster_uuid;
        continue;
    }
    if (*peer_cluster_uuid != reply.cluster_uuid) {
        ... error
        failed = true;
    }
}
```

Generally all the scopes make it a bit tricky to follow, and a bit error prone in that sense. IMO chipping away at "failure" cases with `continue` statements helps readability. Again though this is just cosmetic, so feel free to punt.